xarray/core/dataarray.py

```diff
@@ -3880,16 +3879,16 @@ def pad(
           * x        (x) float64 nan 0.0 1.0 nan
           * y        (y) int64 10 20 30 40
             z        (x) float64 nan 100.0 200.0 nan
-        >>> da.pad(x=1, constant_values=np.nan)
+
+        Careful, ``constant_values`` can be coerced to the data type of the array which
+        may lead to a loss of precision:
```

Comment:
is this unconditional?
```suggestion
        Careful, ``constant_values`` are coerced to the data type of the array which
        may lead to a loss of precision:
```